fe/fe-core/src/main/java/com/starrocks/persist/Storage.java

```diff
@@ -210,6 +210,23 @@ public void writeFrontendRoleAndNodeName(FrontendNodeType role, String nameNode)
         }
     }
 
+    public void writeFeStartFeHostTypeAndHost(String hostType, String host) throws IOException {
+        Preconditions.checkState(!Strings.isNullOrEmpty(hostType));
+        Preconditions.checkState(!Strings.isNullOrEmpty(host));
+        Properties properties = new Properties();
+        properties.setProperty("hostType", hostType);
+        properties.setProperty("host", host);
+        writeFrontendRoleAndNodeName(this.role, this.nodeName);
+        try (RandomAccessFile file = new RandomAccessFile(new File(metaDir, ROLE_FILE), "rws")) {
+            file.seek(file.length());
+            try (FileOutputStream out = new FileOutputStream(file.getFD())) {
+                properties.store(out, null);
+                file.setLength(out.getChannel().position());
+            }
+            file.close();
```

Comment:
file will be closed after the try block ends, because it is generated in `try ()`.
You don't need to call the close explicitly.

Reply:
Why write host to the role file, but you do not use it?